modules/tftp/server.rb

```diff
@@ -145,8 +145,18 @@ def self.fetch_boot_file dst, src
     # Ensure that our image directory exists
     # as the dst might contain another sub directory
     FileUtils.mkdir_p destination.parent
+    choose_protocol_and_fetch src
+  end
 
-    ::Proxy::HttpDownload.new(src.to_s, destination.to_s).start
+  def self.choose_protocol_and_fetch(src)
+    case src.downcase.split(':').first
```

Comment:
I think ```URI(src).scheme``` is a bit better here: if anything, no custom uri parsing and therefore no need for tests.